### PR TITLE
isolate ebpf-based check types

### PR DESCRIPTION
**Pull request by brycekahle**
### What does this PR do?

Isolates the types shared between `agent` and `system-probe` for ebpf-based checks. 

### Motivation

Prevents agent/process-agent from importing third-party packages they have no need for.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by hmahmood — APPROVED** [review]

**Review by kkhor-datadog — APPROVED** [review]

File: cmd/system-probe/modules/oom_kill_probe.go [hunk 1]
```diff
@@ -29,7 +29,7 @@ var OOMKillProbe = module.Factory{
 	ConfigNamespaces: []string{},
 	Fn: func(cfg *config.Config) (module.Module, error) {
 		log.Infof("Starting the OOM Kill probe")
-		okp, err := probe.NewOOMKillProbe(ebpf.NewConfig())
+		okp, err := oomkill.NewOOMKillProbe(ebpf.NewConfig())
```
Comment (L3n41c) [inline cmd/system-probe/modules/oom_kill_probe.go:32]: As the name of the check (`oomkill`) is now part of the package name, would it make sense to simplify the name of the function ?

```suggestion
		okp, err := oomkill.NewProbe(ebpf.NewConfig())
```

File: cmd/system-probe/modules/oom_kill_probe.go [hunk 2]
```diff
@@ -43,7 +43,7 @@ var OOMKillProbe = module.Factory{
 var _ module.Module = &oomKillModule{}
 
 type oomKillModule struct {
-	*probe.OOMKillProbe
+	*oomkill.OOMKillProbe
```
Comment (L3n41c) [inline cmd/system-probe/modules/oom_kill_probe.go:46]:
```suggestion
	*oomkill.Probe
```

File: cmd/system-probe/modules/tcp_queue_tracer.go [hunk 3]
```diff
@@ -27,7 +27,7 @@ var TCPQueueLength = module.Factory{
 	Name:             config.TCPQueueLengthTracerModule,
 	ConfigNamespaces: []string{},
 	Fn: func(cfg *config.Config) (module.Module, error) {
-		t, err := probe.NewTCPQueueLengthTracer(ebpf.NewConfig())
+		t, err := tcpqueuelength.NewTCPQueueLengthTracer(ebpf.NewConfig())
```
Comment (L3n41c) [inline cmd/system-probe/modules/tcp_queue_tracer.go:30]: As the name of the check (`tcpqueuelength`) is now part of the package name, would it make sense to simplify the name of the function ?

```suggestion
		t, err := tcpqueuelength.NewTracer(ebpf.NewConfig())
```

File: cmd/system-probe/modules/tcp_queue_tracer.go [hunk 4]
```diff
@@ -42,7 +42,7 @@ var TCPQueueLength = module.Factory{
 var _ module.Module = &tcpQueueLengthModule{}
 
 type tcpQueueLengthModule struct {
-	*probe.TCPQueueLengthTracer
+	*tcpqueuelength.TCPQueueLengthTracer
```
Comment (L3n41c) [inline cmd/system-probe/modules/tcp_queue_tracer.go:45]:
```suggestion
	*tcpqueuelength.Tracer
```

**Review by val06 — APPROVED** [review]

**Review by L3n41c — APPROVED** [review]
LGTM 👍🏻

I only left a tiny remark about the fact that, as some functions and types are now in a package named from the check type (`oomkill`, `tcpqueuelength`), we might want to remove it from the functions and types name to avoid a verbose duplication of information.

**Pull request merged**